subiquity/server/mounter.py

```diff
@@ -121,9 +124,17 @@ async def mount(self, device, mountpoint, options=None, type=None):
             opts.extend(['-o', options])
         if type is not None:
             opts.extend(['-t', type])
+        if os.path.exists(mountpoint):
+            create = False
+        else:
+            if os.path.isdir(device):
```

Comment:
Wait a minute. If device is a block device this isn't going to do the right thing at all?

Reply:
I do believe this part is correct, though was inconsistent between here and in the removal.  Please see https://github.com/canonical/subiquity/pull/1673

In local tests, bind mounting a device over a file does seem to DTRT.  My test was to go to a tempdir, touch a file, bind mount a drive to that file, then run `sfdisk -J` against the bind mount source and destination and get reasonable results.